lib/memoizable/memory.rb

```diff
@@ -7,11 +7,13 @@ class Memory
 
     # Initialize the memory storage for memoized methods
     #
+    # @param [Hash] memory
+    #
     # @return [undefined]
     #
     # @api private
-    def initialize
-      @memory  = ThreadSafe::Cache.new
+    def initialize(memory)
+      @memory  = memory
```

Comment:
__TODO:__ This refactoring can be split up into smaller chunks.

For example, the injection of the parameter could happen on it's own, and to start it could inject `ThreadSafe::Cache.new`.